client/packages/common/src/ui/components/inputs/DatePickers/BaseDatePickerInput/BaseDatePickerInput.tsx

```diff
@@ -1,16 +1,42 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import {
   DatePicker,
   DatePickerProps,
 } from '@mui/x-date-pickers/DatePicker/DatePicker';
 import { BasicTextInput } from '../../TextInput/BasicTextInput';
 import { StandardTextFieldProps, TextFieldProps } from '@mui/material';
 import { useAppTheme } from '@common/styles';
+import { DateUtils } from '@common/intl';
+import { useDebounceCallback } from '@common/hooks';
 
 export const BaseDatePickerInput: FC<
-  Omit<DatePickerProps<Date, Date>, 'renderInput'>
+  Omit<DatePickerProps<Date, Date>, 'renderInput' | 'value'> & {
+    value: Date | string | null;
+  }
 > = props => {
   const theme = useAppTheme();
+  const [internalValue, setInternalValue] = useState<Date | null>(null);
+
+  useEffect(() => {
+    // This sets the internal state from parent when first loading (i.e. when
+    // the internal date is still empty)
+    if (props.value && internalValue === null)
+      setInternalValue(DateUtils.getDateOrNull(props.value));
+  }, [props.value]);
+
+  const isInvalid = (value: Date | string | number | null | undefined) => {
+    const dateValue =
+      typeof value === 'string' ? DateUtils.getDateOrNull(value) : value;
+    return !!value && !DateUtils.isValid(dateValue);
+  };
```

Comment:
Checks for invalid dates, but doesn't return "invalid" when value is empty (so input doesn't show error state before any input)